src/client.ts

```diff
@@ -0,0 +1,132 @@
+import * as request from "request-promise-native";
+
+interface assoc {
+    [key: string]: any;
+}
+
+export class Client {
+
+    endpoint: string = 'https://appwrite.io/v1';
+    headers: assoc = {
+        'content-type': '',
+        'x-sdk-version': 'appwrite:typescript:',
+    };
+    
+    /**
+     * Set Project
+     *
+     * Your project ID
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setProject(value: string): this {
```

Comment:
As a general note, no need to specify the return type explicitly, most of the time.
TS usually infers it pretty well from the implementation, and you're just writing more code for the same result.

Some project prefer it that way to make things explicit, so it's really up to you. I'm in the less-typing camp personally.

---

This _wouldn't_ compile anyway though (since `number` is not assignable to type `string`):
```ts
function foo(): string {
  return 1;
}
```

so it's not like putting an explicit return type wins if there's an actual conflict between them.